sdk/clientcore/http-netty4/src/main/java/io/clientcore/http/netty4/implementation/Netty4StreamingHttp2Adapter.java

```diff
@@ -0,0 +1,129 @@
+            throw connectionError(PROTOCOL_ERROR, "Data Frame received for unknown stream id %d", streamId);
+        }
+
+        int dataReadableBytes = data.readableBytes();
+        data = Unpooled.copiedBuffer(data);
+        if (endOfStream) {
+            ctx.fireChannelRead(new DefaultLastHttpContent(data));
+        } else {
+            ctx.fireChannelRead(new DefaultHttpContent(data));
+        }
+
+        // All bytes have been processed.
+        return dataReadableBytes + padding;
+    }
+
+    @Override
+    public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers headers, int padding,
+        boolean endOfStream) throws Http2Exception {
+        onHeadersRead(ctx, streamId, headers, -1, (short) -1, false, padding, endOfStream);
+    }
+
+    @Override
+    public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers headers, int streamDependency,
+        short weight, boolean exclusive, int padding, boolean endOfStream) throws Http2Exception {
+        Http2Stream stream = connection.stream(streamId);
+        if (stream == null) {
+            throw connectionError(PROTOCOL_ERROR, "Header Frame received for unknown stream id %d", streamId);
+        }
+
+        HttpHeaders httpHeaders = new WrappedHttp11Headers(new io.clientcore.core.http.models.HttpHeaders());
+        addHttp2ToHttpHeaders(streamId, headers, httpHeaders, HttpVersion.HTTP_1_1, false, false);
+
+        HttpResponse response;
+        if (endOfStream) {
+            response = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, parseStatus(headers.status()),
+                Unpooled.EMPTY_BUFFER, new WrappedHttp11Headers(new io.clientcore.core.http.models.HttpHeaders()),
+                new WrappedHttp11Headers(new io.clientcore.core.http.models.HttpHeaders()));
+            addHttp2ToHttpHeaders(streamId, headers, (FullHttpMessage) response, false);
+        } else {
+            HttpHeaders wrappedHeaders = new WrappedHttp11Headers(new io.clientcore.core.http.models.HttpHeaders());
+            response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, parseStatus(headers.status()), wrappedHeaders);
+            addHttp2ToHttpHeaders(streamId, headers, wrappedHeaders, HttpVersion.HTTP_1_1, false, false);
+        }
+
+        // Add special headers for stream dependency and weight.
+        if (streamDependency > Http2CodecUtil.CONNECTION_STREAM_ID) {
+            response.headers()
+                .setInt(HttpConversionUtil.ExtensionHeaderNames.STREAM_DEPENDENCY_ID.text(), streamDependency);
+        }
+        if (weight > 0) {
+            response.headers().setShort(HttpConversionUtil.ExtensionHeaderNames.STREAM_WEIGHT.text(), weight);
+        }
+
+        ctx.fireChannelRead(response);
+    }
+
+    @Override
+    public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorCode) throws Http2Exception {
+        throw Http2Exception.streamError(streamId, Http2Error.valueOf(errorCode),
```

Comment:
Should we clean up any remaining message in the stream before throwing here on reset?